# HG changeset patch
# User Pantxo Diribarne <pantxo.diribarne@gmail.com>
# Date 1443366749 -7200
#      Sun Sep 27 17:12:29 2015 +0200
# Node ID 88459dd6780fb04562a7f7f203ecc14da8e7b5f8
# Parent  818122703f1e27c0d90249b83b85e250b2132326
Fix inconsistent toolbar state after creating an uicontrol (bug #46025)

* Figure.cc (Figure::eventNotifyAfter): when a child has been created, use update method to properly decide whether to show the toolbar.

diff --git a/libgui/graphics/Figure.cc b/libgui/graphics/Figure.cc
--- a/libgui/graphics/Figure.cc
+++ b/libgui/graphics/Figure.cc
@@ -729,19 +729,17 @@ Figure::eventNotifyAfter (QObject* watch
               updateBoundingBox (true, UpdateBoundingBoxSize);
               break;
 
             case QEvent::ChildAdded:
               if (dynamic_cast<QChildEvent*> (xevent)->child
                   ()->isWidgetType())
                 {
                   gh_manager::auto_lock lock;
-                  const figure::properties& fp = properties<figure> ();
-
-                  showFigureToolBar (! hasUiControlChildren (fp));
+                  update (figure::properties::ID_TOOLBAR);
                 }
 
             default:
               break;
             }
         }
       else if (watched == m_menuBar)
         {
